### PR TITLE
DR-1167 Add 400 error handling behaviour

### PR DESCRIPTION
**Pull request by mvillaseco**
# Overview
 Add error handling for 400 errors.

# Changes
 - Add new module "detached--error" for cases when we have to show an error not related to an input.
 - Detached--error works with max-height making posible to show future errors with a nice transition.
 - Add error handling

# In progress 
 - Test

### CODE REVIEW

File: src/wwwroot/services/settings.js [hunk 1]
```diff
@@ -139,14 +140,15 @@
...
-    function billingPayment(agreement) {
+    function billingPayment(agreement, onExpectedError) {
       var url = RELAY_CONFIG.baseUrl
         + '/accounts/'
         + auth.getAccountName()
         + '/agreements';
 
       return $http({
         actionDescription: 'action_billing_payment',
+        tryHandleError: function(rejection){ return tryHandleError(rejection, onExpectedError); },
```
Comment (andresmoschini) [inline src/wwwroot/services/settings.js]: Please do not reuse domain handle error validation for agreements

**Comment by mvillaseco** [comment]
Is this good to go?

File: src/wwwroot/services/settings.js [hunk 2]
```diff
@@ -139,14 +146,15 @@
...
-    function billingPayment(agreement) {
+    function billingPayment(agreement, onExpectedError) {
       var url = RELAY_CONFIG.baseUrl
         + '/accounts/'
         + auth.getAccountName()
         + '/agreements';
 
       return $http({
         actionDescription: 'action_billing_payment',
+        tryHandleError: function(rejection){ console.log("1");console.log(rejection); return tryHandleErrorBilling(rejection, onExpectedError); },
```
Comment (andresmoschini) [inline src/wwwroot/services/settings.js]: Please, clean this line.

**Comment by andresmoschini** [comment]
👍

**Pull request merged**